rpcs3/Emu/RSX/VK/VKFragmentProgram.cpp

```diff
@@ -228,43 +228,17 @@ void VKFragmentDecompilerThread::insertMainStart(std::stringstream & OS)
 		{
 			if (m_prog.front_back_color_enabled)
 			{
-				if (PI.name == "spec_color")
+				if (PI.name == "spec_color" && m_prog.back_color_specular_output && m_prog.front_color_specular_output)
```

Comment:
This approach is similar to the DX12 one with the modification to use back color output that introduced by kd-11 .